openlibrary/core/models.py

```diff
@@ -217,6 +218,10 @@ def _get_d(self):
             "l": self._get_lists_cached(),
         }
 
+    def get_key_numeric(self):
+        """Returns just the numeric part of the key."""
+        return int(re.search(r'\d+', self.key))
```

Comment:
It's used here to choose the lower of two OL IDs when identifiers are otherwise the same: https://github.com/internetarchive/openlibrary/pull/10092/files#diff-8a66754753640315d80bf708c3483a13439e24736f081161b22e2d59cee76314R183 

I haven't tried yet if this will work with just a string, which was going to be part of the unit tests I add when I mark this PR as ready for review

Reply:
But why this method instead of [`openlibrary.utils.extract_numeric_id_from_olid()`](https://github.com/internetarchive/openlibrary/blob/0dfc2fafc096823dc1ed6268e30bdbca65ce3240/openlibrary/utils/__init__.py#L168-L181) for this? What does this method do that the existing function doesn’t do?

Reply:
probably nothing, I've just never seen that function before